### PR TITLE
refactor: inline gas setting ui provider

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-2270

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/contexts/UITaskManager.tsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
-import type { InjectedDialogProps } from './components/index.js'
 
 export interface ContextOptions<T, R> {
     show(options?: Omit<T, 'open'>, signal?: AbortSignal): Promise<R>
 }
 
-export interface BaseDialogProps<T> extends Pick<InjectedDialogProps, 'open' | 'onClose'> {
+export interface BaseModalPopperProps<T> {
+    open: boolean
+    onClose?(event: {}, reason: 'backdropClick' | 'escapeKeyDown'): void
```
Comment (hanszhoou) [inline packages/shared/src/contexts/UITaskManager.tsx:23]: Mui `DialogProps` and `MenuProps` have the same `onClose()`.

File: packages/shared/src/UI/components/SelectGasSettingsToolbar/index.tsx [hunk 2]
```diff
@@ -223,6 +160,8 @@ export function SelectGasSettingsToolbarUI({
         return formatUSD(formatWeiToEther(gasFee).times(nativeTokenPrice))
     }, [gasFee, nativeTokenPrice])
 
+    const selectGasSettingsMenu = useGasSettingsMenu(NetworkPluginID.PLUGIN_EVM)
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/SelectGasSettingsToolbar/index.tsx]: Don't dead-code to EVM.

File: packages/shared/src/UI/components/SelectGasSettingsToolbar/index.tsx [hunk 3]
```diff
@@ -223,6 +160,8 @@ export function SelectGasSettingsToolbarUI({
...
+
     return gasOptions && !isZero(gasFee) ? (
         <Box className={classes.section}>
             <Typography className={classes.title}>Gas fee</Typography>
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/SelectGasSettingsToolbar/index.tsx]: i18n

**Comment by guanbinrui** [comment]
Please fix unresolved comments here: https://github.com/DimensionDev/Maskbook/pull/7463.

File: packages/shared/src/contexts/UITaskManager.tsx [hunk 4]
```diff
@@ -46,6 +62,27 @@ export const createUITaskManager = <TaskOptions extends BaseDialogProps<Result>,
...
+                        // when the essential content of currentTarget would be closed over,
+                        //  we can set the anchorEl with currentTarget's bottom sibling to avoid it.
+                        const finalAnchor = options?.anchorSibling
+                            ? (element.nextElementSibling as HTMLElement) ?? undefined
+                            : element
+                        setAnchorEl(finalAnchor)
+                        // HACK: it seems like anchor doesn't work correctly
+                        // but a force repaint can solve the problem.
+                        window.requestAnimationFrame(update)
+                    }
```
Comment (UncleBill) [inline packages/shared/src/contexts/UITaskManager.tsx:84]: Let's keep `TaskManagerProvider` generic, move these into SelectGasSettingsMenu.

**Comment by hanszhoou** [comment]
> Please fix unresolved comments here: https://github.com/DimensionDev/Maskbook/pull/7463.

Done

File: packages/shared/src/contexts/UITaskManager.tsx (hunk 4 above)
Comment (hanszhoou) [reply packages/shared/src/contexts/UITaskManager.tsx:84]: I think `TaskManagerProvider` is not generic, it only serves `<InjectedDialog />` before this PR.

File: packages/mask/src/components/InjectedComponents/PostActions.tsx [hunk 5]
```diff
@@ -12,6 +13,7 @@ const ActionsRenderer = createInjectHooksRenderer(
 
 export function PostActions({ isFocusing }: { isFocusing?: boolean }) {
     const identifier = usePostInfoDetails.author()
+    const [disabled, setDisabled] = useState(false)
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/PostActions.tsx:16]: unused state.

File: packages/theme/src/Components/Checkbox/index.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import type { FC } from 'react'
+import { Icons } from '@masknet/icons'
+import type { RadioProps } from '../Radio'
+
+export interface CheckboxProps extends RadioProps {}
+
+export const Checkbox: FC<CheckboxProps> = ({ checked, ...rest }) => {
+    const Icon = checked ? Icons.CheckboxChecked : Icons.CheckboxUnchecked
+    return <Icon role="button" {...rest} />
+}
```
Comment (Jack-Works) [inline packages/theme/src/Components/Checkbox/index.tsx:10]: don't do this.

File: packages/theme/src/Components/Radio/index.tsx [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+import type { FC } from 'react'
+
+export interface RadioProps extends GeneratedIconProps {
+    checked?: boolean
+}
+
+export const Radio: FC<RadioProps> = ({ checked, ...rest }) => {
+    const Icon = checked ? Icons.RadioButtonChecked : Icons.RadioButtonUnChecked
+    return <Icon role="button" {...rest} />
+}
```
Comment (Jack-Works) [inline packages/theme/src/Components/Radio/index.tsx:11]: don't do this.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

**Comment by hanszhoou** [comment]
Replaced by https://github.com/DimensionDev/Maskbook/pull/7548

**Pull request closed without merging**